### ISSUE TITLE
ES6 class doesn't recognize super

### ISSUE THREAD

**Issue by mvolkmann** [eslint/eslint#1968]
For the following code:
````
export class Foo {
}

export class Bar extends Foo {
  constructor() {
    super();
  }
}
````

the console output is:
````
eslint-demo2.js
  6:4  error  'super' is not defined  no-undef

✖ 1 problem (1 error, 0 warnings)
````

**Comment by nzakas** [comment]
Could be escope.

**Comment by nzakas** [comment]
Yup, escope bug: https://github.com/estools/escope/issues/50

**Comment by nzakas** [comment]
Working on this. Going to hack in a fix until escope is updated.

**Comment by BerkeleyTrue** [comment]
I am getting issues with super and blocked-scope-vars rule, is this related or a separate issue?
```js
  constructor(map) {
    super();
    this._map = map;

    this._observers = [];
    this._startAction = new Rx.Subject();
    this._endAction = new Rx.Subject();

    Rx.Observable.call(this, function (observer) {
      this._observers.push(observer);
      return {
        dispose: function () {
          var index = this._observers.indexOf(observer);
          if (index !== -1) {
            this._observers.splice(index, 1);
          }
        }
      };
    });
  }
```
![1__tmux](https://cloud.githubusercontent.com/assets/6775919/6813462/97ea5754-d232-11e4-8b8c-a0296d9bef80.png)


eslint 0.17.1

```json
    "superInFunctions": true,
    "classes": true,
    "block-scoped-var": 2,
```

**Comment by nzakas** [comment]
It is related to the same escope issue. Will go away when we upgrade.

**Comment by BerkeleyTrue** [comment]
Sweet. thanks for the quick response!

**Comment by nzakas** [comment]
Fixed by #2235
